QGA-5 create record model

Bekhzod96 opened the pull request:
Record model

Bekhzod96 committed:
REFERANCE: QGA-5
DESCRIPTION: RECORD MODEL CREATED

Bekhzod96 committed:
REFERANCE:QGA-5, DECRIPTION: RECORD ROUTER HAS BEEN CREATED AND CONNECTED TO CONTROLLER

Bekhzod96 committed:
REFERANCE: QGA-5
DESCRIPTION: Record Model has been created and modified and added
 additional validation

Gruseckis commented:
Change target to development

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,27 @@
+    recordID: { $in: [mongoose.Types.ObjectId(recordId.map())] }
+  });
+
+export { save, getRecordById, updateById, deleteRecordById, getAllRecords };
```
You need to export all functions make sure the names are correct.

Bekhzod96 replied:
I exported it properly

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,27 @@
+
+const getRecordsFromIdArray = async recordId =>
+  RefordModel.find({
+    recordID: { $in: [mongoose.Types.ObjectId(recordId.map())] }
```
This won't work. 

- You need to supply key that you will be making query against like _id.
- You are using map function but you do not know how it works. At the end of the day you won't supply array to $in it will be only one object with function inside of it. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Gruseckis commented on routes/recordsRoute.js:
```diff
@@ -0,0 +1,17 @@
+import express from "express";
```
This should not be a part of this pull request

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,27 @@
+
+const save = async model => new RecordModel(model).save();
+const getAllRecords = async () => RecordModel.find();
+const getRecordById = async _id => RecordModel.findById({ _id });
```
You can just pass this function _id without casting it to object

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,27 @@
+  RecordModel.findByIdAndUpdate(recordId, update, { new: true });
+
+const deleteRecordById = async recordId =>
+  RecordModel.findByIdAndRemove(recordId);
```
use findByIdAndDelete()

sadinovsr commented on models/recordsModel.js:
```diff
@@ -0,0 +1,27 @@
+const recordSchema = new mongoose.Schema(
+  {
+    questionId: { type: String, required: true, unique: false },
+    answers: [{ type: Number, requred: true }]
```
There is a typo for 'required'

Bekhzod96 replied:
Thank you

Bekhzod96 committed:
REFERANCE:QGA-5 create record model
DESCRIPTION: I have fixed all mistakes but I am not sure about
findAllById in Array.

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,38 @@
+const deleteRecordById = async recordId =>
+  RecordModel.findByIdAndDelete(recordId);
+
+const getRecordsFromIdArray = async _id => {
```
name of the parameter should be meaningful. idArray.
you need to map idArray to be an array of mongoose.Types.ObjectId with id in it
then you need to pass that array into find function with flag $in.

Bekhzod96 committed:
REFERANCE:QGA-5 create record model
DESCRIPTION: I have fixed findAllIdFromArray

Bekhzod96 committed:
REFERANCE:QGA-5-create-record-model
DESCRIPTION: Answear type has been changed to String

Gruseckis commented on models/recordsModel.js:
```diff
@@ -0,0 +1,35 @@
+const deleteRecordById = async recordId =>
+  RecordModel.findByIdAndDelete(recordId);
+
+const getRecordsFromIdArray = async arrId => {
```
Are you a pirate? Arrrrrr? Change to meaningful name :)

Bekhzod96 committed:
REFERANCE:QGA-5-create-record-model
DESCRIPTION: ArrId has been changed to ArrayID
